vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java

```diff
@@ -67,7 +67,7 @@ public class Popover extends Component implements HasAriaLabel, HasComponents,
     private static Integer defaultHideDelay;
     private static Integer defaultFocusDelay;
     private static Integer defaultHoverDelay;
-    private static boolean uiInitListenerRegistered = false;
+    static boolean uiInitListenerRegistered = false;
```

Comment:
Changed the visibility here to be able to reset the flag on each test since for the variable being static, its value is preserved in between tests.

Reply:
Would it make sense to use an `AtomicBoolean` here and update the if check like the following?
```Java
if (uiInitListenerRegistered.compareAndSet(false, true)) {
    VaadinService.getCurrent().addUIInitListener(e -> applyConfigurationForUI(e.getUI()));
}
```

It would get rid of `synchronized` methods, keep the synced block minimal, and still keep the thread safety.

Reply:
Yeah, this also seems to work and requires fewer changes, so I think it might be better. Changed.